reconcile/external_resources/manager.py

```diff
@@ -0,0 +1,330 @@
+        state.resource_hash = r.resource_hash
+        state.reconciliation = r
+        self.state_mgr.set_external_resource_state(state)
+
+    def _need_secret_sync(
+        self, r: Reconciliation, state: ExternalResourceState
+    ) -> bool:
+        if r.action == Action.APPLY and state.resource_status == ResourceStatus.CREATED:
+            return True
+        return False
+
+    def _sync_secrets(self, keys: Iterable[ExternalResourceKey]) -> None:
+        specs = [spec for key in keys if (spec := self.er_inventory.get(key))]
+        self.secrets_reconciler.sync_secrets(specs=specs)
+
+    def _build_external_resource(
+        self, spec: ExternalResourceSpec, er_inventory: ExternalResourcesInventory
+    ) -> ExternalResource:
+        f = self.factories.get_factory(spec.provision_provider)
+        resource = f.create_external_resource(spec)
+        f.validate_external_resource(resource)
+        return resource
+
+    def _serialize_resource_input(self, resource: ExternalResource) -> str:
+        return json.dumps(
+            resource.dict(exclude={"data": {FLAG_RESOURCE_MANAGED_BY_ERV2}})
+        )
+
+    def handle_resources(self) -> None:
+        desired_r = self._get_desired_objects_reconciliations()
+        deleted_r = self._get_deleted_objects_reconciliations()
+        to_sync: set[ExternalResourceKey] = set()
+        for r in desired_r.union(deleted_r):
+            state = self.state_mgr.get_external_resource_state(r.key)
+            need_sync = self._update_in_progress_state(r, state)
+            if need_sync:
+                to_sync.add(r.key)
+
+            metrics.set_gauge(
+                ExternalResourcesReconcileErrorsGauge(
+                    provision_provider=r.key.provision_provider,
+                    provisioner_name=r.key.provisioner_name,
+                    provider=r.key.provider,
+                    identifier=r.key.identifier,
+                ),
+                float(state.reconciliation_errors),
+            )
+
+            if self._resource_needs_reconciliation(reconciliation=r, state=state):
+                self.reconciler.reconcile_resource(reconciliation=r)
+                self._update_state(r, state)
+
+            if to_sync:
+                self._sync_secrets(keys=to_sync)
+
+    def handle_dry_run_resources(self) -> None:
+        desired_r = self._get_desired_objects_reconciliations()
+        deleted_r = self._get_deleted_objects_reconciliations()
+        reconciliations = desired_r.union(deleted_r)
+        triggered = set[Reconciliation]()
+
+        for r in reconciliations:
+            state = self.state_mgr.get_external_resource_state(key=r.key)
+            if (
+                r.action == Action.APPLY and state.resource_hash != r.resource_hash
+            ) or r.action == Action.DESTROY:
+                triggered.add(r)
+
+        threaded.run(self.reconciler.reconcile_resource, triggered, thread_pool_size=5)
```

Comment:
`thread_pool_size` hard code to 5?

Reply:
I don't think we need more than that for now. We can implement a better pool size handling in the future, but as a starting point I think we are good with 5.

Reply:
we have `@threaded()` in other integrations to make it customizable as cli arg, can be `@threaded(default=5)`.

Reply:
Implemented the thread pool size with threaded.